### DISCUSSION TITLE
I'm training VITS. Characters None not found. showing but all Characters are present  in my character_config .then why giving me this error ??

### DISCUSSION THREAD

**Discussion by omkarade** [coqui-ai/TTS#2034] [General Q&A]
TRAINING (2022-09-28 17:32:10)
[None, 'औ', None, 'र', None, ' ', None, 'अ', None, 'प', None, 'न', None, 'े', None, ' ', None, 'व', None, 'ि', None, 'द', None, '्', None, 'य', None, 'ा', None, 'र', None, '्', None, 'थ', None, 'ि', None, 'य', None, 'ो', None, 'ं', None, ' ', None, 'क', None, 'ी', None, ' ', None, 'स', None, 'फ', None, 'ल', None, 'त', None, 'ा', None, ' ', None, 'म', None, 'े', None, 'ं', None, ' ', None, 'आ', None, 'न', None, 'ं', None, 'द', None, ' ', None, 'प', None, 'ा', None, 'न', None, 'े', None, ' ', None, 'क', None, 'े', None, ' ', None, 'उ', None, 'न', None, 'क', None, 'े', None, ' ', None, 'त', None, 'र', None, 'ी', None, 'क', None, 'े', None, ' ', None, 'क', None, 'ो', None]
[!] Character None not found in the vocabulary. Discarding it.
/usr/local/lib/python3.7/dist-packages/torch/functional.py:607: UserWarning: stft will soon require the return_complex parameter be given for real inputs, and will further require that return_complex=True in a future PyTorch release. (Triggered internally at ../aten/src/ATen/native/SpectralOps.cpp:800.)

up showing all Characters are present in my character_config then why this showing me those Characters None not found[]

**Comment by WeberJulian** [comment]
You need to give us more information if you want some help. Start by explaining what you are doing training/fine-tuning, explain the dataset and upload config/recipe.

**Reply by omkarade** [reply]
audio_config = BaseAudioConfig(
    sample_rate=22050,
    do_trim_silence=True,
    trim_db=60.0,
    signal_norm=False,
    mel_fmin=0.0,
    mel_fmax=8000,
    spec_gain=1.0,
    log_func="np.log",
    ref_level_db=20,
    preemphasis=0.0,
)
character_config=CharactersConfig(
  pad='-',
  eos='~',
  bos='^',
  characters='अआइईउऊएऐओऔऑअंअःऋकखगघङचछजझञटठडढणतथदधनपफबभमयरलवशषसहषतरज्ञफ़ख़ज़ऍळज़ढ़क़ड़ ्ॉ ा ि ी ु ू ृृ ेे ैै ो ौ ़ँॅे',
  punctuations= "–.:;,'‘’“”()!?।*",
  phonemes='None', 
) 
config = VitsConfig(
    audio=audio_config,
    run_name="vits_ljspeech",
    batch_size=32,
    eval_batch_size=16,
    batch_group_size=5,
    num_loader_workers=0,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="hindi_cleaners",
    use_phonemes=False,
    dashboard_logger='tensorboard', 
    save_step=8000, 
    distributed_backend='nccl',
    distributed_url='tcp://localhost:54321',
    compute_input_seq_cache=False,
    characters=character_config,
    print_step=250,
    print_eval=False,
    mixed_precision=False,
    output_path=output_path,
    datasets=[dataset_config],
)

**Reply by omkarade** [reply]
@WeberJulian I'm training VITS model

**Comment by p0p4k** [comment]
What is the issue here? None is just getting discarded, it's a side effect of text formatter.

**Reply by omkarade** [reply]
so i need change anything??or keep it as it is ?

**Reply by p0p4k** [reply]
Just keep it as it is.

**Reply by omkarade** [reply]
I need change anything in cofig to train fast model??

**Reply by p0p4k** [reply]
Invest in better GPU.

**Comment by dostee-hemin** [comment]
Set `add_blank` to false in the config.json file like this:

`"add_blank": false,`

**Comment by Ataullha** [comment]
That doesn't create any problem. Just Ignore it.
